lib/sendgrid/email.ex

```diff
@@ -193,6 +196,22 @@ defmodule SendGrid.Email do
   end
 
   @doc """
+  Sets an custom header.
+
+      Email.put_header(%Email{}, "HEADER_KEY", "HEADER_VALUE")
+
+  """
+  @spec put_header(Email.t, String.t, String.t) :: Email.t
+  def put_header(%Email{} = email, header_key, header_value) do
```

Comment:
Can you rename your new function to `add_header`? `add_x` has been the convention for modifying lists in this lib.